### PR TITLE
Document the `is_index` property of page objects

### PR DESCRIPTION
**Pull request by IreneKnapp**
I was told on https://github.com/mkdocs/mkdocs/issues/3882 that this is likely to be a stable interface and that there's no reason it shouldn't be documented as such, so I'm submitting this PR to do that.

Thanks in advance!

### CODE REVIEW

**Comment by IreneKnapp** [comment]
apologies for the force-pushes, I inadvertently included my NixOS packaging of the various mkdocs plugins as part of the original commit

File: mkdocs/structure/pages.py [hunk 1]
```diff
@@ -124,6 +124,7 @@ def active(self, value: bool):
 
     @property
     def is_index(self) -> bool:
+        """When `True`, indicates that this page is an index page, in the sense that it gets sorted to the beginning of the list of its siblings because its filename was `index.md` or something similar."""
```
Comment (pawamoy) [inline mkdocs/structure/pages.py:127]: Feels a bit long for a summary.

```suggestion
        """When `True`, indicates that this page is an index page.
        
        Index pages get sorted to the beginning of the list of their siblings
        because their filename is `index.md` or `README.md`.
        """
```

Actually not entirely sure about `README.md`. I think @tomchristie would be able to confirm (and maybe rephrase to something better).

**Review by pawamoy — CHANGES_REQUESTED** [review]
No worries, and thanks! Just one comment.

File: mkdocs/structure/pages.py (hunk 1 above)
Comment (pawamoy) [reply mkdocs/structure/pages.py:127]: Maybe just drop the second line in the body.

**Comment by reteps** [comment]
Unrelated, but `is_index` currently could have improved handling. See https://github.com/squidfunk/mkdocs-material/issues/7561
